SEP-0009.md

```diff
@@ -0,0 +1,69 @@
+| status        | discussion                                |
+| discussion    | https://github.com/sunpy/sunpy/issues/993 |
+
+# Introduction
+
+This SEP proposes that SunPy transitions all internal, and external use of a
+Python object to represent a date or time from the standard library `datetime`
+module to the classes provided in `astropy.time`. The main classes being
+replacing `datetime.date` and `datetime.datetime` with `astropy.Time` and
+replacing `datetime.timedelta` with `astropy.time.TimeDelta`.
+
+# Detailed Description
+
+Since its inception the SunPy core library has used `datetime` for handling
+time. This is a very useful standard library module, however it lacks some key
+benefits that are useful to the solar community:
+
+- Support for non-UTC time scales. This is especially useful for JSOC and HMI data which use the TAI time scale rather than UTC. 
+- Support for high (double-double) precision time representation and arithmetic.
+- Support for leap seconds.
+- Support for different representations of time, i.e. Julian Day, unix time, GPS.
+- Support for reading and writing the new FITS Time standard.
+- Support for light travel time calculations for different locations on Earth.
+- Support for custom Time scales and formats, i.e. utime.
+
+
+## parse_time
+
+All over SunPy when time-like objects are accepted as input, this input is
+sanitised with the `sunpy.time.parse_time` function. This function currently
+returns a `datetime.datetime` object. The major API change proposed by the
+transition to `astropy.time` is the return value of the `parse_time` function
+will become `astropy.time.Time`. It is also proposed that the function signature
+of `parse_time` be standardised with the signature of `astropy.time.Time` for
+clarity, the details of this are left to the implementation.
+
+
+## Other Changes
+
+Many other parts of SunPy will need to be adapted to work with `astropy.time`
+and convert to datetime in as few places as possible to maintain the advantages
+of the Astropy representation. This will result in the return values of any
+functions returning time objects also becoming `astropy.time.Time` objects. It
+is not expected that there will be any significant changes in functionality.
+
+
+## sunpy.timeseries
+
+It should be noted that because `sunpy.timeseries.GenericTimeseries` is using
+Pandas it will not be changed to use `astropy.time` it would be necessary to
+migrate from Pandas to `astropy.table` to gain the advantages of `astropy.time`.
+Timeseries data are currently not sufficiently supported by `astropy.table` this
+is proposed to be added in [APE 9](https://github.com/astropy/astropy-APEs/pull/12).
+
+The API of the `sunpy.timeseries` module will be converted to use `astropy.time`
+as appropriate although the actual temporal index of the data (which is the
+pandas dataframe) will not be changed.
+
+# Decision Rationale
+This is a great idea because it's already being implemented. ;)
```

Comment:
Something about why now.  "SunPy 1.0 is intended to stabilize the API.  This SEP will introduce a stable implementation of time that already works with Astropy code.  SunPy relies on Astropy code in many places, and this change will therefore make SunPy code maintenance simpler."

Reply:
This section is to be written after acceptance. If anyone has comments on this wording we can adapt it and add it after approval.

Reply:
Ok, my comment is not part of the decision to adopt, but rather a reason to adopt.